javascript/ql/lib/semmle/javascript/frameworks/Hapi.qll

```diff
@@ -14,16 +14,24 @@ module Hapi {
       // `server = new Hapi.Server()`
       this = DataFlow::moduleMember("hapi", "Server").getAnInstantiation()
       or
-      // server = Glue.compose(manifest, composeOptions)
+      // `server = Glue.compose(manifest, composeOptions)`
       this = DataFlow::moduleMember("@hapi/glue", "compose").getAnInvocation()
       or
-      // server inside a plugin
-      // TODO match `function (server, options)`
+      // `register (server, options)`
       exists(Function f |
         this.(DataFlow::ParameterNode).getParameter() = f.getParameter(0) and
+        f.getName() = "register" and
         f.getParameter(0).getName() = "server" and
         f.getParameter(1).getName() = "options"
       )
```

Comment:
If you want to keep this, then it belongs in something similar to a `RouteHandlerCandidate`.  
(Which you can then promote by importing `AdditionalRouteHandlers.qll`)
It's too much of a heuristic to merge into main.